src/batch/src/executor/table_function.rs

```diff
@@ -52,32 +50,11 @@ impl TableFunctionExecutor {
     async fn do_execute(self: Box<Self>) {
         let dummy_chunk = DataChunk::new_dummy(1);
 
-        let data_chunk_type = {
-            match self.table_function.return_type() {
-                DataType::Struct(s) => s.fields.clone(),
-                other => vec![other],
-            }
-        };
-
-        let mut data_chunk_builder = DataChunkBuilder::new(data_chunk_type, self.chunk_size);
-
-        for array in self.table_function.eval(&dummy_chunk).await? {
-            let len = array.len();
-            if len == 0 {
-                continue;
-            }
-            let data_chunk = match array.as_ref() {
-                ArrayImpl::Struct(s) => DataChunk::from(s),
-                _ => DataChunk::new(vec![Column::new(array.clone())], len),
-            };
-
-            for chunk in data_chunk_builder.append_chunk(data_chunk) {
-                yield chunk;
-            }
-        }
-
-        if let Some(chunk) = data_chunk_builder.consume_all() {
-            yield chunk;
+        #[for_await]
+        for chunk in self.table_function.eval(&dummy_chunk).await {
+            let chunk = chunk?;
+            // remove the first column
+            yield chunk.split_column_at(1).1;
```

Comment:
Should we also consider `Struct` here? (or is this only for UDF and we've already handled there?)